api/v1beta2/network_types.go

```diff
@@ -94,12 +104,56 @@ var (
 
 // TargetGroupHealthCheck defines health check settings for the target group.
 type TargetGroupHealthCheck struct {
-	Protocol        *string `json:"protocol,omitempty"`
-	Path            *string `json:"path,omitempty"`
-	Port            *string `json:"port,omitempty"`
-	IntervalSeconds *int64  `json:"intervalSeconds,omitempty"`
-	TimeoutSeconds  *int64  `json:"timeoutSeconds,omitempty"`
-	ThresholdCount  *int64  `json:"thresholdCount,omitempty"`
+
+	// The protocol to use to health check connect with the target.
+	// Defaults to TCP.
+	// +kubebuilder:default=TCP
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// The destination for health checks on the targets when using the protocol HTTP or HTTPS.
+	// Defaults to /redyz.
```

Comment:
```suggestion
	// Defaults to /readyz.
```

Reply:
can you use this default with TCP protocol.

Reply:
I think it would make sense to make the default HTTP wouldn't it?

Reply:
@faiq yeah, I think it is the most correct path. But I am afraid of changing defaults in CAPA.
We are using HTTPS by default on OpenShift.

+ @vincepri thoughts?

Reply:
Having a default here makes it non-backward compatible. Why not leave these values empty for end users to setup instead?

Reply:
@vincepri considering HTTP* is a best practice, specially for the graceful shutdown: should we need to plan in the future and emit some kind message to use this in new installs?

Reply:
we can try to default on create only, rather than default all the time?

Reply:
@vincepri Good question. I am new in CAPA project to provide a good answer. :)
I am suggesting that possibility for CAPA considering the AWS and KAS recommendation. But imo for "downstream" projects, all they need is to be allowed to customize the health check attributes (addressed in #4849 , for now blocked by this PR).

Anyway this PR is not changing the default, so we can move that thread to an issue if you think it is something to be considered.